snuba/query/expressions.py

```diff
@@ -139,6 +143,45 @@ def accept(self, visitor: ExpressionVisitor[TVisited]) -> TVisited:
         return visitor.visitColumn(self)
 
 
+@dataclass(frozen=True)
+class SubscriptableReference(Expression):
+    """
+    Accesses one entry of a subscriptable column (for example key based access on
```

Comment:
I can see us expanding this if we expand the capabilities in the query language like supporting sequences access, allowing expressions in the key, or building something similar for accessing structures.
This is the bare minimum for what the query language allows.